src/dist/failure_detector/failure_detector.cpp

```diff
@@ -340,37 +339,42 @@ void failure_detector::end_ping(::dsn::error_code err, const beacon_ack& ack, vo
 
     master_map::iterator itr = _masters.find(node);
 
-    if ( itr == _masters.end() )
+    if (itr == _masters.end())
     {
         dwarn("received beacon ack without corresponding master, ignore it, "
-              "remote_master[%s], local_worker[%s]",
-              node.to_string(), primary_address().to_string());
-
-        return;
+            "remote_master[%s], local_worker[%s]",
+            node.to_string(), primary_address().to_string());
+        return false;
     }
 
     master_record& record = itr->second;
     if (!ack.allowed)
     {
         dwarn("worker rejected, stop sending beacon message, "
-              "remote_master[%s], local_worker[%s]",
-              node.to_string(), primary_address().to_string());
+            "remote_master[%s], local_worker[%s]",
+            node.to_string(), primary_address().to_string());
         record.rejected = true;
-        return;
+        return false;
     }
 
-    if (is_time_greater_than(beacon_send_time, record.last_send_time_for_beacon_with_ack))
+    if (!is_time_greater_than(beacon_send_time, record.last_send_time_for_beacon_with_ack))
```

Comment:
In that case, b2 actually helps us save the worker back, i.e., the worker is still connected to the master. Or am I wrong somewhere?